### ISSUE TITLE
Use CI to build all tools in this repository on every commit.

### ISSUE THREAD

**Issue by kwoodson** [openshift/azure-misc#85]
There were some recent changes to directory structure which broke some of our tools including the publishing tools.

We need CI to ensure that our repo is sane.

**Comment by mjudeikis** [comment]
fix: https://github.com/openshift/azure-misc/pull/84

**Comment by kwoodson** [comment]
@mjudeikis, #84 was merged.  Does that PR fix this issue?  If so, can we close?

**Comment by jim-minter** [comment]
@kwoodson please clarify the description - CI should run builds of all tools in this repo on every commit.

**Comment by kwoodson** [comment]
Clarified.

**Comment by mjudeikis** [comment]
yes, it got merged. I kinda messed up a little bit with #84 (see comments in the PR) but yes, it fixed the issue.

**Comment by jim-minter** [comment]
reopening - we don't check bindata generation

**Comment by kargakis** [comment]
We can also have a simple e2e test where all SRE apps get deployed and we
expect them to be running. We could potentially setup a cluster to run
continuously to avoid the churn involved with install and teardown but not
necessary.

On Oct 12, 2018 9:02 PM, "Jim Minter" <notifications@github.com> wrote:

Reopened #85 <https://github.com/openshift/azure-misc/issues/85>.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/azure-misc/issues/85#event-1901475766>, or mute
the thread

**Comment by mjudeikis** [comment]
> reopening - can you confirm we check bindata generation? 

no, we dont. we should. So we need to add bindata generation and validation on built containers.

**Comment by kargakis** [comment]
We should also add a job that vets changes in the prometheus rules. Here is the job I had added to do the same thing for our CI prom rules.
https://github.com/openshift/release/blob/17d2b89150c939f07742d20bf70e5bd7439060bc/ci-operator/jobs/openshift/release/openshift-release-master-presubmits.yaml#L3-L23
https://github.com/openshift/release/tree/master/projects/prometheus/test

**Comment by mjudeikis** [comment]
This sounds like separate story for me

**Comment by kargakis** [comment]
Yes, I was using this issue as a grab bag for CI in this repo. Opened https://github.com/openshift/azure-misc/issues/114 and https://github.com/openshift/azure-misc/issues/115
